File: backend/Services/Parsers/Crosswordle2Parser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class Crosswordle2Parser : BasicScoreResultParser
    {
        private ILogger<Crosswordle2Parser> _logger;

        public Crosswordle2Parser(ILogger<Crosswordle2Parser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string GameName => "Crosswordle 2";
        public override string? HelpText => "Guess a two word phrase.";
        protected override Regex Parser => new Regex($"^Crosswordle \\d+ .*? (?<{ScoreGroup}>[\\d]+)/");
        protected override string? ExtraContent => Url;
        public override string Url => "https://crosswordle.serializer.ca/";
    }
}

File: lambdas/src/parsers/crosswordle2-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class Crosswordle2Parser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Crosswordle 2';
  readonly helpText = 'Guess a two word phrase.';
  readonly url = 'https://crosswordle.serializer.ca/';
  protected readonly extraContent = this.url;
  
  // Match "Crosswordle 123 ... 5/6" - score is before the slash
  protected readonly parser = new RegExp(
    `^Crosswordle \\d+ .*? (?<${BasicScoreResultParser.SCORE_GROUP}>[\\d]+)/`,
    'm'
  );
}
